modules/apps/commerce/commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/servlet/taglib/AddToCartTag.java

```diff
@@ -152,6 +154,13 @@ public int doStartTag() throws JspException {
 							 commerceContext.getCommerceSiteType()));
 				}
 			}
+
+			int commerceOrderTypesCount =
```

Comment:
why don't you check the type of the channel?

Reply:
Because I also need to check if there is more than 1 order type to open the modal.
I added this validation on my work in progress
```
int commerceOrderTypesCount = _commerceOrderTypeLocalService.getCommerceOrderTypesCount(PortalUtil.getCompanyId(httpServletRequest), CommerceChannel.class.getName(), _commerceChannelId, true);

_showOrderTypeModal = commerceOrderTypesCount > 1 && CommerceAccountConstants.SITE_TYPE_B2B == commerceContext.getCommerceSiteType();
```